### PR TITLE
convert to ts

### PR DESCRIPTION
**Pull request by zaher-aa**
### converted client file structure to typescript

### CODE REVIEW

**Review by khaled-samy — DISMISSED** [review]

File: client/package.json [hunk 1]
```diff
@@ -1,33 +1,32 @@
...
     "@emotion/react": "^11.9.0",
     "@emotion/styled": "^11.8.1",
-    "@fontsource/poppins": "^4.5.5",
+    "@fontsource/poppins": "^4.5.8",
     "@mui/icons-material": "^5.6.2",
-    "@mui/material": "^5.6.2",
-    "axios": "^0.26.1",
+    "@mui/material": "^5.6.4",
+    "@mui/styled-engine-sc": "^5.6.1",
+    "@types/jest": "^27.5.0",
```
Comment (amjed-98) [inline client/package.json:11]: shouldn't this be listed in devDependencies? 
in most cases anything that relates to typescript should be listed in devDependencies.

this article should help you decide:
https://stackoverflow.com/questions/45176661/how-do-i-decide-whether-types-goes-into-dependencies-or-devdependencies

File: client/package.json [hunk 2]
```diff
@@ -1,33 +1,32 @@
...
+    "@types/node": "^16.11.33",
```
Comment (amjed-98) [inline client/package.json:12]: do you really need @types/node? , because we are working on the browser here

File: client/package.json [hunk 3]
```diff
@@ -1,33 +1,32 @@
...
+    "axios": "^0.27.2",
     "prop-types": "^15.8.1",
-    "react": "^18.0.0",
-    "react-dom": "^18.0.0",
+    "react": "^18.1.0",
+    "react-dom": "^18.1.0",
     "react-router-dom": "^6.3.0",
     "react-scripts": "5.0.1",
-    "uuid": "^8.3.2"
+    "styled-components": "^5.3.5",
```
Comment (amjed-98) [inline client/package.json:21]: do we really need this? I think material UI is enough, what do you think?

**Review by amjed-98 — CHANGES_REQUESTED** [review]

File: client/.gitignore [hunk 4]
```diff
@@ -1,2 +1,23 @@
...
+# misc
+.DS_Store
+.env.local
+.env.development.local
+.env.test.local
+.env.production.local
+
+npm-debug.log*
+yarn-debug.log*
+yarn-error.log*
```
Comment (amjed-98) [inline client/.gitignore:23]: maybe just ignore the files we actually have

**Review by amjed-98 — CHANGES_REQUESTED** [review]

**Review by khaled-samy — APPROVED** [review]

**Review by amjed-98 — APPROVED** [review]

**Review by Mu7ammadAbed — APPROVED** [review]

**Pull request merged**